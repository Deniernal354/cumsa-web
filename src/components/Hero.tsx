import Image from "next/image";
import Link from "next/link";

type Props = {
  title: string;
  subtitle?: string;
  cta?: { label: string; href: string }[];
  image?: { src: string; alt: string };
  overlay?: boolean;
};

export default function Hero({ title, subtitle, cta, image, overlay = true }: Props) {
  return (
    <section className="relative overflow-hidden">
      {image && (
        <div className="absolute inset-0 -z-10">
          <Image src={image.src} alt={image.alt} fill priority className="object-cover" />
          {overlay && <div className="absolute inset-0 bg-[var(--navy)]/60" />}
        </div>
      )}
      <div className={`container-base ${image ? "py-24 md:py-32" : "py-12 md:py-16"} text-white`}>
        <h1 className="text-4xl md:text-6xl font-extrabold tracking-tight">{title}</h1>
        {subtitle && <p className="mt-4 max-w-2xl text-white/85 text-lg">{subtitle}</p>}
        {cta && cta.length > 0 && (
          <div className="mt-8 flex flex-wrap gap-3">
            {cta.map((c) => (
              <Link
                key={c.href}
                href={c.href}
                className="px-5 py-2.5 rounded-md bg-[var(--blue)] text-white transition-base hover:opacity-90"
              >
                {c.label}
              </Link>
            ))}
          </div>
        )}
      </div>
    </section>
  );
}
