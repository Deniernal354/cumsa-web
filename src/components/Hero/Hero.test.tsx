import { render } from "@testing-library/react";
import Hero from "./Hero";

test("Hero snapshot", () => {
  const { asFragment } = render(<Hero title="Test Title" subtitle="Test Subtitle" />);
  expect(asFragment()).toMatchSnapshot();
});
