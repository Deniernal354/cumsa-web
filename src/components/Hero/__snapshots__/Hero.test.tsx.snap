// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero snapshot 1`] = `
<DocumentFragment>
  <section
    class="relative overflow-hidden z-0"
    style="z-index: 1;"
  >
    <div
      class="container-base py-12 md:py-16 text-white relative"
      style="z-index: 1;"
    >
      <h1
        class="text-4xl md:text-6xl font-extrabold tracking-tight"
      >
        Test Title
      </h1>
      <p
        class="mt-4 max-w-2xl text-white/85 text-lg"
      >
        Test Subtitle
      </p>
    </div>
  </section>
</DocumentFragment>
`;
